### PR TITLE
List team members by UserId

### PR DESCRIPTION
**Pull request by fisx**
see https://github.com/zinfra/backend-issues/issues/1197

### CODE REVIEW

File: services/galley/src/Galley/API.hs [hunk 1]
```diff
@@ -162,6 +162,28 @@ sitemap = do
...
+  post "/teams/:tid/get-members-by-ids-using-post" (continue Teams.bulkGetTeamMembersH) $
+    zauthUserId
+      .&. capture "tid"
+      .&. def (unsafeRange hardTruncationLimit) (query "maxResults")
+      .&. jsonRequest @UserIdList
+      .&. accept "application" "json"
+  document "POST" "bulkGetTeamMembers" $ do
+    summary
+      "Get team members by user id list\n\
+      \NOTE: `has_more` field in response body is always `false`."
```
Comment (mheinzel) [inline services/galley/src/Galley/API.hs:174]:
```suggestion
    summary "Get team members by user id list"
    notes "The `has_more` field in the response body is always `false`."
```

File: services/galley/src/Galley/API/Teams.hs [hunk 2]
```diff
@@ -342,6 +344,25 @@ getTeamMembers zusr tid maxResults = do
...
+  pure . json $ teamMemberListJson withPerms memberList
+
+-- | like 'getTeamMembers', but with an explicit list of users we are to return.
+bulkGetTeamMembers :: UserId -> TeamId -> Range 1 HardTruncationLimit Int32 -> [UserId] -> Galley (TeamMemberList, TeamMember -> Bool)
+bulkGetTeamMembers zusr tid maxResults uids = do
+  unless (length uids <= fromIntegral (fromRange maxResults)) $
+    throwM bulkGetMemberLimitExceeded
+  mems <- Data.teamMembersLimited tid uids
+  Data.teamMember tid zusr >>= \case
+    Nothing -> throwM notATeamMember
```
Comment (mheinzel) [inline services/galley/src/Galley/API/Teams.hs:360]: Also applies to the existing `getTeamMembers` above, but why do we fetch the result before checking that `zusr` is a member of the team? Should the query not be moved into the `Just m` branch?

File: services/galley/test/integration/API/Teams.hs [hunk 3]
```diff
@@ -210,6 +214,34 @@ testListTeamMembersTruncated = do
       "member list does not indicate that there are more members"
       (listFromServer ^. teamMemberListHasMore)
 
+testListTeamMembersDefaultLimitByIds :: TestM ()
+testListTeamMembersDefaultLimitByIds = do
+  (owner, tid, [member1, member2]) <- Util.createBindingTeamWithNMembers 2
+  check owner tid [owner, member1, member2]
+  check owner tid [member1, member2]
+  check owner tid [member1]
+  check owner tid [] -- a bit silly, but hey.
```
Comment (mheinzel) [inline services/galley/test/integration/API/Teams.hs:223]: How should the endpoint behave when you also ask for a non-existing user ID. Or *only* for non-existing ones? Or if your `maxResults`  is above the truncation limit?
We could test for some of these cases, too. (imho, mostly the first one case would be useful, maybe also the second)

File: services/galley/test/integration/API/Teams.hs [hunk 4]
```diff
@@ -210,6 +214,34 @@ testListTeamMembersTruncated = do
...
+  where
+    check :: UserId -> TeamId -> [UserId] -> TestM ()
```
Comment (mheinzel) [inline services/galley/test/integration/API/Teams.hs:225]:
```suggestion
    check :: HasCallStack => UserId -> TeamId -> [UserId] -> TestM ()
```

Ah, another small thing.

File: services/galley/src/Galley/API/Teams.hs (hunk 2 above)
Comment (fisx) [reply services/galley/src/Galley/API/Teams.hs:359]: I agree.  26a4bb6530e4a2dedea21144e012f0ee1bb51ac2

File: services/galley/test/integration/API/Teams.hs (hunk 3 above)
Comment (fisx) [reply services/galley/test/integration/API/Teams.hs]: good idea.  31627edc4f5ebf9eaafc33a2c98d558ce4f1f2a6

I had some truncation errors in there, too, but then I realized that's tested in the next case, so I removed them.

**Pull request merged**